plugin-api/src/main/java/org/sonar/api/batch/sensor/issue/ExternalIssue.java

```diff
@@ -50,7 +58,23 @@ public interface ExternalIssue extends IIssue {
 
   /**
    * Type of the issue.
+   * @deprecated since 10.1, use {@link #impacts()} instead
    */
+  @Deprecated(since = "10.1")
+  @CheckForNull
   RuleType type();
 
+  /**
+   * Impacts of the issue.
+   * @since 10.1
+   */
+  Map<SoftwareQuality, org.sonar.api.issue.impact.Severity> impacts();
+
+  /**
+   * Attribute of the issue according to Clean Code Taxonomy.
+   * @since 10.1
+   */
+  @CheckForNull
+  CleanCodeAttribute cleanCodeAttribute();
```

Comment:
Clarification question for me - will the behaviour for `ExternalIssue` be different than for 'normal' `Issue`?  The issues get attributes from the rule, they won't have attributes on their own. Won't it be the case here?

Reply:
Thanks a lot for the answer @benjamin-campomenosi-sonarsource , I'm not very familiar with the `ExternalIssue` :) 
A follow-up question - does it mean that external issues can have a different attribute than the rule they belong to or e.g. two issues belonging to the same rule can have a different attributes? Or is this used just as a `carrier`, and the expectation is that the attribute will be the same per rule?

In SC we also allow to import `ExternalIssue`, so it would be good to be aligned

Reply:
According to me, the issue is just used as a _carrier_, and the expectation is that the attribute will be same per rule. 
However, it's a good question I don't know the current behavior if 2 external issues have a different Type/Severity for the same rule. I'll check in our code to determine the current logic, and raise the point to Gab'. 
I’ll let you know as soon as I know more

Reply:
From a product POV, we still have some adjustments to look at, but from an plugin-api pov we are good. 
As for _adhoc_ rule, it is the external issue that carries the rule, one must have the `attribute` and the `quality`.
For the product results, I’ll share our findings with you once we’ve made our choices. 
Is that okay with you?

Reply:
yes, thanks a lot!